File: src/C#/BehavioralPatterns/Mediator/MultipleExamples/CommunicationHub/Mediators/GroupMediator.cs
using MultipleExamples.CommunicationHub.Participants;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace MultipleExamples.CommunicationHub.Mediators
{
    public class GroupMediator : IMediator
    {
        public List<Group> Groups { get; set; } = new List<Group>();

        public virtual void Notify(Participant sender, object senderArgs)
        {
            var groupsOfParticipant = Groups.Where(gr => gr.DoesParticipantExist(sender)).ToList();

            var receivers = groupsOfParticipant
                .SelectMany(gr => gr.Participants)
                .Where(p => p != sender)
                .Distinct()
                .ToList();

            receivers.ForEach(r => r.Receive(sender, senderArgs));
        }
    }
}

File: src/JavaScript/BehavioralPatterns/Mediator/CommunicationHub/mediators/groupMediator.js

const Mediator = require("./mediator");

class GroupMediator extends Mediator {
  constructor() {
    super();
    this.groups = [];
  }

  get groups() {
    return this._groups;
  }

  set groups(value) {
    this._groups = value;
  }

  notify(sender, senderArgs) {
    const groupsOfParticipant = this.groups.filter(group => group.doesParticipantExist(sender));

    const receivers = groupsOfParticipant
      .flatMap(group => group.participants)
      .filter(participant => participant !== sender)
      .filter((value, index, self) => self.indexOf(value) === index); // Remove duplicates

    receivers.forEach(receiver => receiver.receive(sender, senderArgs));
  }
}

module.exports = GroupMediator;